AElf.OS.Network.Grpc/GrpcNetworkServer.cs

```diff
@@ -0,0 +1,207 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Security.Cryptography;
+using System.Threading.Tasks;
+using AElf.Common;
+using AElf.Kernel.Account;
+using AElf.OS.Network.Grpc.Events;
+using AElf.OS.Network.Temp;
+using Google.Protobuf;
+using Grpc.Core;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Options;
+using Org.BouncyCastle.Utilities.Encoders;
+using Volo.Abp.DependencyInjection;
+using Volo.Abp.EventBus.Local;
+
+namespace AElf.OS.Network.Grpc
+{
+    public class GrpcNetworkServer : IAElfNetworkServer, IPeerAuthentificator, ISingletonDependency
+    {
+        private readonly NetworkOptions _networkOptions;
+        
+        private readonly IAccountService _accountService;
+        private readonly IBlockService _blockService;
+        
+        private Server _server;
+        private readonly List<GrpcPeer> _authenticatedPeers;
+        
+        public ILocalEventBus EventBus { get; set; }
+        public ILogger<GrpcNetworkServer> Logger { get; set; }
+        
+        public GrpcNetworkServer(IOptionsSnapshot<NetworkOptions> options, IAccountService accountService, 
+            IBlockService blockService)
+        {
+            _accountService = accountService;
+            _blockService = blockService;
+            _networkOptions = options.Value;
+            
+            Logger = NullLogger<GrpcNetworkServer>.Instance;
+            EventBus = NullLocalEventBus.Instance;
+            
+            _authenticatedPeers = new List<GrpcPeer>();
+        }
+        
+        public async Task StartAsync()
+        {
+            // todo inject block service
+            var p = new GrpcPeerService(this, _blockService);
```

Comment:
maybe you should implement IPeerAuthentificator in another class ,

you can implement a factory to avoid use new operation like https://github.com/abpframework/abp/blob/55458a7a081f6eb6124d55950d4d61e5a48f5fc7/framework/src/Volo.Abp.EventBus/Volo/Abp/EventBus/IocEventHandlerFactory.cs